### ISSUE TITLE
Add operation functions

### ISSUE THREAD

**Issue by jake-bickle** [AlexanderJDupree/LinkedListsCPP#31]
Operations modify the list by means not as explicit as modifiers. They may change zero, one, or many elements. The LinkedList must have the following functionality: 

- [x] sort : Sort the list
There will be two versions of sort:
Version 1 will have no arguments and utilize the < operator. 
Version 2 will utilize a function pointer.
Ex. sort()
```c++
LinkedList<int> list = {4, 2, 1, 6};
list.sort();  // Results in {1, 2, 4, 6}
```
```c++
Ex. sort(* Compare)
bool reverse(const int i&, const int j&) { return i > j }

LinkedList<int> list = {4, 2, 1, 6};
list.sort(reverse); // Results in {6, 4, 2, 1}
```

- [x] reverse : Reverse the order of the list
Ex. reverse()
```c++
LinkedList<char> list = {'A', 'D', 'B', 'T'};
list.reverse(); // Results in {'T', 'B', 'D', 'A'}
```

- [x] remove : Remove elements with specific value
Ex. sort(const <T>& value)
```c++
LinkedList<char> list = {'B', 'B', 'A', 'C', 'D'};
list.remove('B'); // Results in {'A', 'C', 'D'}
```

- [x] remove_if : Remove elements that fulfill a condition. Takes a pointer function that returns a boolean value.
Ex. sort(* Condition)
```c++
bool isEven(const int& i) { return i % 2 == 0 }

LinkedList<int> list = {2, 3, 4, 5};
list.remove_if(isEven);  // Results in {3, 5}
```

- [x] unique : Remove any duplicate data
Ex. unique()
```c++
LinkedList<int> list = {5, 2, 3, 1, 1, 5, 2, 3, 4, 4};
list.unique(); // Results in {5, 2, 3, 1, 4}
```

- [x] find : Return an iterator to the first element that compares equal to the target value
Ex. find(const_reference target)
```c++
LinkedList<char> list { 'A', 'B', 'C', 'Z', 'E' }
LinkedList<char>::iterator it = list.find('Z');

assert(*it == 'Z');
```

- [x] find_if :Returns an iterator to the first element that returns true from the predicate function.

Ex. find(Predicate pred)

```c++
LinkedList<int> list { 1, 2, 3, 4, 5 }
LinkedList<int>::iterator it = list.find([](const int& val) { return val >= 3; })

assert (*it == 3);
```

**Comment by AlexanderJDupree** [comment]
Reverse implemented in PR #43 
Unique implemented in PR #44

**Comment by AlexanderJDupree** [comment]
Remove and remove_if implemented in PR #45

**Comment by AlexanderJDupree** [comment]
find and find_if implemented in PR #49

**Comment by AlexanderJDupree** [comment]
sort implemented in PR #50
